src/System.Resources.Extensions/src/System/Resources/Extensions/PreserializedResourceWriter.cs

```diff
@@ -25,14 +33,43 @@ partial class PreserializedResourceWriter
             RuntimeResourceSetFullyQualifiedName :
             ResSetTypeName;
 
+        // a collection of primitive types in a dictionary, indexed by type name
+        // using a comparer which handles type name comparisons similar to what
+        // is done by refelection
+        private static readonly Dictionary<string, Type> s_primitiveTypes = new[]
+        {
+            typeof(string),
+            typeof(int),
+            typeof(bool),
+            typeof(char),
+            typeof(byte),
+            typeof(sbyte),
+            typeof(short),
+            typeof(long),
+            typeof(ushort),
+            typeof(uint),
+            typeof(ulong),
+            typeof(float),
+            typeof(double),
+            typeof(decimal),
+            typeof(DateTime),
+            typeof(TimeSpan)
+
+            // The following primitive types do not define a conversion from string
+            // typeof(byte[]),
+            // typeof(Stream)
+        }.ToDictionary(t => t.FullName, TypeNameComparer.Instance);
+
         /// <summary>
-        /// Adds a resource of specified type represented by a string value which will be 
+        /// Adds a resource of specified type represented by a string value.
+        /// If the type is a primitive type 
+        /// which will be 
```

Comment:
nit: per our offline discussion, would be nice if you add one more comment explaining why such API's not taking a type converter and instead taking just a type name.